### ISSUE TITLE
IO Exception Updates.XML

### ISSUE THREAD

**Issue by kb2010** [gephi/gephi#2429]
I see this in console n startup and loading new saved file. 

Figured out this stops me from retrieving plugins. It says Unable to connect. However, tested the proxy connection in the Proxy configuration window and it passed.

Opening 0.9.2 on the side, the same plugin update works fine.  although the proxy connection doesnt. which is odd. 

basically 0.9.2 even though proxy does not test fine I can still dl the plugins. on 0.9.3 the proxy connects fine but i cant download the plugins.

This is in 0.9.3 - Snapshot

````
[WARNING] Not all children in Menu/View/ marked with the position attribute: [org-netbeans-core-windows-actions-ToggleFullScreenAction.instance], but some are: [ToggleFullScreenAction.shadow] 

[INFO] Reading URL https://raw.githubusercontent.com/gephi/gephi-plugins/gh-pages/plugins/0.9.3-SNAPSHOT/updates.xml failed (java.io.FileNotFoundException: https://raw.githubusercontent.com/gephi/gephi-plugins/gh-pages/plugins/0.9.3-SNAPSHOT/updates.xml) 

[INFO] null 
java.io.IOException: https://raw.githubusercontent.com/gephi/gephi-plugins/gh-pages/plugins/0.9.3-SNAPSHOT/updates.xml
	at org.netbeans.modules.autoupdate.updateprovider.DownloadListener.notifyException(DownloadListener.java:78)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogCache.copy(AutoupdateCatalogCache.java:221)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogCache.writeCatalogToCache(AutoupdateCatalogCache.java:74)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogProvider.refresh(AutoupdateCatalogProvider.java:134)
	at org.netbeans.modules.autoupdate.services.UpdateUnitProviderImpl.refresh(UpdateUnitProviderImpl.java:165)
	at org.netbeans.api.autoupdate.UpdateUnitProvider.refresh(UpdateUnitProvider.java:171)
	at org.netbeans.modules.autoupdate.ui.actions.AutoupdateCheckScheduler$4.run(AutoupdateCheckScheduler.java:130)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1418)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:278)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)

Caused by: java.io.FileNotFoundException: https://raw.githubusercontent.com/gephi/gephi-plugins/gh-pages/plugins/0.9.3-SNAPSHOT/updates.xml
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$9.run(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$9.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.AccessController.doPrivilegedWithCombiner(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$2.call(NetworkAccess.java:156)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$2.call(NetworkAccess.java:130)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

Caused by: java.io.FileNotFoundException: https://raw.githubusercontent.com/gephi/gephi-plugins/gh-pages/plugins/0.9.3-SNAPSHOT/updates.xml
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$9.run(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$9.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.AccessController.doPrivilegedWithCombiner(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.base/java.net.HttpURLConnection.getResponseCode(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(Unknown Source)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task.checkRedirect(NetworkAccess.java:181)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task.access$800(NetworkAccess.java:60)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$2.call(NetworkAccess.java:148)
	... 5 more
````

**Comment by kb2010** [comment]
OK figured it out. The Update Site needs editing. Not sure if this is correct but go into Tools -> Plugins -> Settings

Then need to edit the URL for the existing items or add a new one. I Linked as follows but i'm not sure if its correct especially using 0.9.2 plugin in 0.9.3.

For Plugins i used: https://raw.githubusercontent.com/gephi/gephi-plugins/gh-pages/plugins/0.9.2/updates.xml
For Updates I used:  https://raw.githubusercontent.com/gephi/gephi/gh-pages/0.9.3-SNAPSHOT/netbeans_site/updates.xml 

The issue is for plugins in gh-Pages/Plugins there is no folder for 0.9.3. so have to revert to 0.9.2.

If someone could confirm what the correct link is that would be could. This isn't a bug or issue with the application though. :)

![image](https://user-images.githubusercontent.com/6481997/153896960-fac5f5b8-d23c-4bc4-a2d7-7653b032e28f.png)

**Comment by kb2010** [comment]
ok one above is done there are a couple of issues installing plugins not sure if this is due to compatibility checks. Is there a way to bypass?
![image](https://user-images.githubusercontent.com/6481997/153900560-95032fdd-c334-428f-aefa-26f5745c10d0.png)
![image](https://user-images.githubusercontent.com/6481997/153900587-87527536-8307-4614-8ba8-4688b9d73af2.png)

**Comment by mbastian** [comment]
Hi, plugins aren't supposed to work at the moment in SNAPSHOT versions so what you're seeing is totally normal. 

This is now resolved as we've released the ([0.9.3](https://github.com/gephi/gephi/releases/tag/v0.9.3))version. Plugins are there as usual. Thank you!
